planner/core/stats.go

```diff
@@ -37,10 +37,10 @@ func (p *basePhysicalPlan) StatsCount() float64 {
 func (p *LogicalTableDual) DeriveStats(childStats []*property.StatsInfo, selfSchema *expression.Schema, childSchema []*expression.Schema) (*property.StatsInfo, error) {
 	profile := &property.StatsInfo{
 		RowCount:    float64(p.RowCount),
-		Cardinality: make([]float64, selfSchema.Len()),
+		Cardinality: make(map[int64]float64, selfSchema.Len()),
```

Comment:
Would it cause performance issues after using map?

Reply:
Probably, but IMO the cost of `map` from `slice` is not that significant and is worth handling code friendly in the future. (Let's do a `/bench`)